src/function/maybe_changed_after.rs

```diff
@@ -228,8 +219,24 @@ where
         false
     }
 
+    /// Validates this memo if it is a provisional memo. Returns true for non provisional memos or
+    /// if the provisional memo has been successfully marked as verified final.
+    #[inline]
+    pub(super) fn validate_may_be_provisional(
+        &self,
+        db: &C::DbView,
+        zalsa: &Zalsa,
+        database_key_index: DatabaseKeyIndex,
+        memo: &Memo<C::Output<'_>>,
+    ) -> bool {
+        // Wouldn't it be nice if rust had an implication operator ...
+        // may_be_provisional -> validate_provisional
+        !memo.may_be_provisional() || self.validate_provisional(db, zalsa, database_key_index, memo)
```

Comment:
Would it be just as good to not add this additional wrapper, and instead just check `memo.may_be_provisional` inside `validate_provisional`? This is the only call site of `validate_provisional`.

Reply:
This seems to regress perf, presumably due to inlining changes